sync_diff_inspector/config/config.go

```diff
@@ -97,6 +98,18 @@ type DataSource struct {
 	// SourceType string `toml:"source-type" json:"source-type"`
 }
 
+// IsAutoSnapshot returns true if the tidb_snapshot is expected to automatically
+// be set from the syncpoint from the target TiDB instance.
+func (d *DataSource) IsAutoSnapshot() bool {
+	return strings.EqualFold(d.Snapshot, "auto")
```

Comment:
Perhaps it would be better to use words such as "auto-cdc", because we also have `dm` and `binlog`.

Reply:
I prefer auto, but yes this does very specifically relate to the CDC use case. The concern I have with `auto-cdc` is two word tokens are more subject to misconfiguration (autocdc, auto_cdc etc).

Reply:
OK. I think it is acceptable to use `auto`, but we also need to update the document and annotation of configuration template at the same time.

I will update the document after the PR is merged. Could you add the annotation to explain the meanings of value `auto` in the follow configuration in this PR?
https://github.com/pingcap/tidb-tools/blob/1d2f00da8c3e934f3c13332cbd3a1981eb8c1887/sync_diff_inspector/config/config.toml#L30-L32